pymodbus/server/simulator/http_server.py

```diff
@@ -300,15 +300,18 @@ async def handle_html(self, request):
 
     async def handle_json(self, request):
         """Handle api registers."""
-        page_type = request.path.split("/")[-1]
-        params = await request.post()
-        json_dict = self.generator_html[page_type][0].copy()
-        result = self.generator_json[page_type][1](params, json_dict)
-        return web.Response(text=f"json build: {page_type} - {params} - {result}")
+        endpoint = request.path.split("/")[-1]
+        params = await request.json()
+        try:
+            result = self.generator_json[endpoint](params)
```

Comment:
There are only one "endpoint" == restapi, which you defined earlier...I think you are mixing endpoints and commands.

Reply:
Yes. I often get confused with the terminology.

I'll change it to `command =`. You also had a previous suggestion on using "request" (which I missed/forgot). Let me know if you prefer that one over command.

Reply:
Changed